### DISCUSSION TITLE
How control page-breaking with three groups in Excel .

### DISCUSSION THREAD

**Discussion by FZOGHLAMI** [eclipse-birt/birt#1260] [Q&A]
Hello,
My report returns more than 1,048,576 rows ( Max limit in XLSX)
I want 1,048,576 rows in sheet 0 . and remainning rows in next sheet ( sheet 1 or tab 1)
How I can configure this With Birt?
if current rows equal 1,048,576 row , I break page ?

_You recommend to control page-breaking (=switch to new sheet for Excel emitter) with the groups themselves, e.g. to set page-break-before: always (or always except first) for one of the groups.
See the page break dialog for the groups._

But the client wishes to display ****the total** (group 3) in the same tab.**
if I break in group 1, I can't ensure the total in the same page ?

### RPTDESIGN :

![B1](https://user-images.githubusercontent.com/24314808/223149954-6c2c04cc-ce32-4da1-a10c-f33e54a9b3ae.png)

### The Good break page with Total 

![image](https://user-images.githubusercontent.com/24314808/223152409-d38772df-9f50-4f75-adf3-d76852ea36c5.png)

### The Bad break page without Total 

![B3](https://user-images.githubusercontent.com/24314808/223153154-8c05e409-51bc-404c-93d8-094ec49430ef.png)

**Comment by hvbtup** [comment]
[page_break.zip](https://github.com/eclipse/birt/files/10900233/page_break.zip)

I created a report demonstrating the issue for you.

If you generate the report as PDF, the page-break-inside: avoid is working.
This is what I would recommend normally.

But it seems this is not working for Excel output.

Unless someone else comes up with a different idea, I think it will just not work.

**Comment by speckyspooky** [comment]
I don't know if the topic is relevant.
But perhaps another solution would be do handle the information of the totals in an additional row
and the count of current used rows. In the extrem case that XLSX-with maximum 1,048,576 will be used
then exactly this line would be the total line.

I used the base report page_break.zip and added an additional total line.
The line number and total calculation will be done in very small scripts and the "pink" row includes an according visibility condition for display an reset of line numbers.

Based on the example each data-row 100 get an additional detail row withthere total.
Perhaps it can help you in one of your next topics.


Screen of additional total line based on line counter:
![2023-04-10 17_23_46-page_break_calc xlsx - Excel](https://user-images.githubusercontent.com/41593722/230932317-31b02814-18f3-4589-9227-07aee4436dca.png)

Example of additional total line based on line counter:
[page_break_calc.zip](https://github.com/eclipse/birt/files/11191489/page_break_calc.zip)
